store/subscription_store/store.go

```diff
@@ -0,0 +1,83 @@
+package store
+
+import (
+	"regexp"
+	"time"
+
+	"github.com/go-errors/errors"
+	"github.com/guregu/null"
+	"github.com/jinzhu/gorm"
+
+	"github.com/torinos-io/api/type/model"
+)
+
+var emailPattern = regexp.MustCompile(`^(?i:[^ @"<>]+|".*")@(?i:[a-z1-9.])+.(?i:[a-z])+$`)
+
+type concreteStore struct {
+	db *gorm.DB
+}
+
+// Store is an interface for CRUD category records
+type Store interface {
+	CreateSubscription(userID null.Int, email string, projectUUID string) (*model.Subscription, error)
+	DeleteSubscription(userID null.Int, projectUUID string) error
+}
+
+// New creates a store
+func New(db *gorm.DB) Store {
+	return &concreteStore{
+		db: db,
+	}
+}
+
+func (s *concreteStore) CreateSubscription(userID null.Int, email string, projectUUID string) (*model.Subscription, error) {
+	subscription := &model.Subscription{}
+
+	finder := s.db.
+		Where("user_id = ?", userID.Int64).
+		Where("project_uuid", projectUUID).
+		Find(subscription)
+
+	if err := finder.Error; err != nil && err != gorm.ErrRecordNotFound {
+		return subscription, errors.Wrap(err, 0)
+	}
+
+	if !emailPattern.MatchString(email) {
```

Comment:
Validation logic like this should not be included in a repository layer. It's an application logic.
Hence all data that flows into the repository must be valid (except for relations and other db-related stuffs).

Let's introduce https://github.com/asaskevich/govalidator, and validate in a service layer.